admin-notice/class-admin-notice-controller.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Automattic\VIP\Admin_Notice;
+
+// function sample_admin_notice__error() {
+//     $class = 'notice notice-info';
+//     $message = __( 'Irks! An error has occurred.', 'sample-text-domain' );
+
+//     printf( '<div class="%1$s"><p>%2$s</p></div>', esc_attr( $class ), esc_html( $message ) );
+// }
+// add_action( 'admin_notices', 'sample_admin_notice__error' );
+
+
+require_once __DIR__ . '/class-admin-notice.php';
+
+
+class Admin_Notice_Controller {
+	private $notice_class = 'notice notice-info';
+	public $all_notices = [];
+
+	public static function init() {
+		$instance = new Admin_Notice_Controller();
```

Comment:
```suggestion
		$instance = new static();
```

No need to repeat the class name. `new self()` wouldn't support classes that extend this one, and since there's no `final` on the class, you're presumably OK with allowing extended classes.

Reply:
As per your advice, I am no longer creating an instance here, so I will resolve this comment.